test/unit/test_auth_okta.py

```diff
@@ -216,6 +216,8 @@ def get_one_time_token():
     # the first time, when step4 gets executed, we return 429
     # the second time when step4 gets retried, we return 200
     def mock_session_request(*args, **kwargs):
+        url = kwargs.get("url")
+        assert url.count("onetimetoken") == 1
```

Comment:
The fix looks good but I don't understand how this test is ensuring that we're looping and the `sso_url` is still good.

Reply:
Agreed. The test should probably check that we don't keep piling on the url?

Reply:
I just updated test to compare the whole url string, it should be clear now